src/axolotl/utils/models.py

```diff
@@ -331,6 +333,14 @@ def load_model(
             model, use_gradient_checkpointing=cfg.gradient_checkpointing
         )
 
+    if cfg.flash_attention:
+        for name, module in model.named_modules():
+            if "norm" in name:
+                module.to(torch_dtype)
+            if "lm_head" in name or "embed_tokens" in name:
+                if hasattr(module, "weight"):
+                    module.to(torch_dtype)
+
```

Comment:
should this be in load_lora? or is flash-attn broken for non-lora trainings w/o this?

Reply:
sorry it looks like i got the indentation wrong here. it was intended to happen after prepare_for_kbit since that's where the issue is coming from.

additionally however maybe we should also check if its a llama based model since i believe (but have not tested) that this is specific to LlamaRMSNorm layers